### PR TITLE
Added flight app created for trial vm demo

### PR DESCRIPTION
**Pull request by rafaelhbarros**
I'm missing the readme.md change which I'm unclear what I need to do.

### CODE REVIEW

**Comment by birdsarah** [comment]
@rafaelhbarros - don't worry about the integration test fail, that's a pytest-selenium versioning mismatch - i'm working on a solution.

**Comment by rafaelhbarros** [comment]
@birdsarah thank you very much.

**Comment by bryevdv** [comment]
@rafaelhbarros since this demo includes python callbacks I don't think the references to `bokeh html` and `bokeh json` are so useful. I'd keep it short and sweet, basically just like this:

https://github.com/bokeh/bokeh/blob/master/examples/app/ohlc/README.md

Which appropriate changes.

**Comment by birdsarah** [comment]
@rafaelhbarros, I took the liberty of cancelling your last travis build. Can you merge the latest master?

**Comment by birdsarah** [comment]
This is a great demo, thanks @rafaelhbarros. 

Some comments:
1) The data will need to be added to our sampledata.download feature - this is pretty straight forward - and I can help with getting it up on S3. Can you confirm that we have permission to redistribute the data?
2) I get an error when I click on Hawaiin Airlines
````
  File "/Users/caged/Dev/bokeh/bokeh/examples/app/flight_delays/main.py", line 55, in on_carrier_change
    update_map()
  File "/Users/caged/Dev/bokeh/bokeh/examples/app/flight_delays/main.py", line 85, in update_map
    size_groups = pd.qcut(df1['arr_flights'].tolist(), len(sizes))
  File "/Users/caged/miniconda3/envs/bokeh_34/lib/python3.4/site-packages/pandas/tools/tile.py", line 169, in qcut
    include_lowest=True)
  File "/Users/caged/miniconda3/envs/bokeh_34/lib/python3.4/site-packages/pandas/tools/tile.py", line 189, in _bins_to_cuts
    raise ValueError('Bin edges must be unique: %s' % repr(bins))
ValueError: Bin edges must be unique: array([   31. ,    31. ,    50.8,    74.8,   481.6,  2257. ])
```
3) I find the demo gets stuck fairly often. If I change the dropdown more than a few times then the plot  stops updating, with the browser console error message: `Bokeh: Error handling message: Error: attempted to retrieve property array for nonexistent field 'color', [object MessageEvent]` - it doesn't seem to be a specific airline that causes it.

File: examples/app/flight_delays/main.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    month = new
+    update_map()
+
+def on_carrier_change(attr, old, new):
+    global carrier
+    carrier = new
+    update_map()
+
+def update_map():
+    global p
```
Comment (bryevdv) [inline examples/app/flight_delays/main.py]: You're not reassigning to `p`, the `global` is not needed

File: examples/app/flight_delays/main.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    adf = adf[adf['country'] == 'United States'][['iata/faa', 'latitude', 'longitude', 'name']]
+    adf = adf[(adf.longitude < max_long) & (adf.longitude > min_long)]
+    adf.set_index('iata/faa', inplace=True)
+
+    return state_xs, state_ys, airline_delays_df, adf
+
+def on_year_change(attr, old, new):
+    global year
+    year = new
+    update_map()
```
Comment (bryevdv) [inline examples/app/flight_delays/main.py]: globals not needed. Just call `update_map` and have `update_map` read the values off the widget objects directly.

File: examples/app/flight_delays/main.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        data = {}
+        data['x'] = []
+        data['y'] = []
+        source.data = data
+        return
+
+    df1['late_pct'] = df1['arr_del15'] / df1['arr_flights'] * 100
+    df1 = df1[np.isfinite(df1.late_pct)]
+
+    colors = list(reversed(Reds5))
```
Comment (bryevdv) [inline examples/app/flight_delays/main.py]: This work doesn't need to be done every update, suggest making it a module scope variable

File: examples/app/flight_delays/main.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+carriers = ['All'] + sorted(airline_delays_df['carrier_name'].unique().tolist())
+airline_selector = Select.create(name='Carrier', value=carrier, options=carriers)
+airline_selector.on_change('value', on_carrier_change)
+
+controls_box = HBox(width=990, children=[year_slider, month_slider, airline_selector])
+
+intro_text = 'The map above shows U.S. airports symbolized by flight delay information for the time period selected above.'
+color_text = 'The color of the airport represents its percentage of delayed flights with lighter red for lower percentage of delays and darker red for greater.'
+size_text = 'The size of the airport represents the total number for flight arrivals.'
+binning_text = 'For both color and size, airports are binned by quintiles.'
```
Comment (bryevdv) [inline examples/app/flight_delays/main.py]: might be nice to put these in a yaml or txt file in the app dir, to demonstrate reading from sidecar resource files.

File: examples/app/flight_delays/main.py (hunk 2 above)
Comment (bryevdv) [inline examples/app/flight_delays/main.py]: For instance, c.f https://github.com/bokeh/bokeh/blob/master/examples/app/sliders.py#L55-L59

**Comment by birdsarah** [comment]
superseded by: #3987

**Pull request closed without merging**